balenaapi/balenaapi.go

```diff
@@ -60,6 +60,8 @@ func GetTagValue(apiKey string, uuid string, tagKey string) (string, error) {
 
 	if len(tags.Data) == 0 {
 		return "", nil
+	} else if len(tags.Data) > 1 {
+		return "", errors.New(fmt.Sprintf("Expected only 1 tag, received %d", len(tags.Data)))
```

Comment:
Potentially dumb question but why is only 1 tag expected? Or when can there be multiple?

Reply:
tag keys are unique per device, so there should only ever be one `MAINTENANCE_WINDOW` tag for that device UUID. there could only be multiple if the filtering api is not working or there's a client-side bug like I had before this fix :) it should never happen in practice but just adding the condition because why not.